src/alos1_2_gcov/alos1_to_gcov.py

```diff
@@ -0,0 +1,502 @@
+import os
+import zipfile
+import subprocess
+import argparse
+from ruamel.yaml import YAML
+from dataclasses import dataclass
+
+@dataclass
+class LatLonBbox:
+    left: float
+    bottom: float    
+    right: float
+    top: float
+
+    def get_as_str(self) -> str:
+        """
+        Return lat/lon bounding box coordinates in a single string,
+        with orientation: [left  bottom  right top].
+        
+        Example: '-156 18.8 -154.7 20.3'
+        """
+        return f"{self.left} {self.bottom} {self.right} {self.top}"
+
+
+def pad_bbox(bbox: LatLonBbox, pad: float = 1.0) -> LatLonBbox:
+    """
+    Get a copy of the input LatLonBbox, but uniformly padded/enlarged.
+
+    Warning: Only correct for North American coordinates.
+    """
+    return LatLonBbox(
+        left=bbox.left - pad,
+        bottom=bbox.bottom - pad,
+        right=bbox.right + pad,
+        top=bbox.top + pad
+    )
+
+
+def compute_union_bbox(ll1: LatLonBbox, ll2: LatLonBbox) -> LatLonBbox:
+    """
+    Compute union of two LotLonBbox objects in N. America.
+
+    Warning: Only correct for North American coordinates.
+    """
+    top = max(ll1.top, ll2.top)
+    bottom = min(ll1.bottom, ll2.bottom)
+
+    left = max(ll1.left, ll2.left)
+    right = min(ll1.right, ll2.right)
+
+    return LatLonBbox(left=left, bottom=bottom, right=right, top=top)
+        
+
+def make_dem(bbox: LatLonBbox, dem_path: str) -> None:
+    """
+    Generate a SRTM DEM Gtiff for the given bounding box.
+    
+    Parameters
+    ----------
+    bbox : LatLonBbox
+        Lat/lon bounding box to get the DEM for.
+    dem_path : str
+        Filepath to the generated DEM Gtiff.
+        Suggested format: "<out_dir>/dem.tif"
+    """
+    # Annoyingly, rasterio cannot find the PROJ_DATA directory
+    # when running in the NASA MAAP ADE.
+    # So, we need to manually set the environment variable, and
+    # then run sardem
+
+    # Step 1: Get the path to PROJ_DATA.
+    #     From Command Line, use the command: echo $PROJ_DATA
+    #     Example outputs:
+    #       In conda base environment in MAAP ADE, this produces: /opt/conda/share/proj
+    #       In a custom conda environment named 'dem', this produces: '/opt/conda/envs/dem/share/proj'
+    result = subprocess.run(['echo $PROJ_DATA'], stdout=subprocess.PIPE, shell=True)
+    proj_data_path = result.stdout.decode('utf-8').strip()
+    
+    os.environ['PROJ_DATA'] = proj_data_path
+    
+    # Step 2: Run sardem
+    os.system(f"sardem --bbox {bbox.get_as_str()} --data-source COP -o {dem_path} --output-format GTiff")
+
+    # Warning: in a Jupyter notebook on NASA MAAP ADE, Steps 1 and 2 must be combined:
+    #     !PROJ_DATA={proj_data_path} sardem --bbox {bbox} --data-source COP -o {dem_file} --output-format GTiff
```

Comment:
Recall how we addressed this in `get_dem`? We don't make a system call. The problem you are facing here is the same one we discussed the other day: the system call does _not_ run within the same conda environment, which is why PROJ_DATA is not set in system call.

```suggestion
    from sardem import cop_dem

    cop_dem.download_and_stitch(dem_path, (bbox.left, bbox.bottom, bbox.right, bbox.top), output_format="GTiff")

    # Warning: in a Jupyter notebook on NASA MAAP ADE, this can be achieved with a system call:
    #     !SAR2D2_ENV={env} ./run.sh sardem --bbox {bbox} --data-source COP -o {dem_file} --output-format GTiff
```

Reply:
Trick is, for expediency we're using the Copernicus DEM b/c accessing it does not require authentication. However, the Copernicus DEM is a newer DEM, whereas the ALOS-1 dataset is older, and it is better to use a DEM that was generated at roughly the same timeframe as the datasets.

Come January, we'll need to update this to use the SRTM DEM, which is more of a contemporary of the ALOS1 mission.

Also, I don't think this suggested command will work; the `env` variable is unknown within the Python module, and `run.sh` is not setup to understand a command of that format.

Reply:
Perhaps you're misreading the suggestion.

I'm suggesting that the function needs only to do this:

```
from sardem import cop_dem
    cop_dem.download_and_stitch(dem_path, (bbox.left, bbox.bottom, bbox.right, bbox.top), output_format="GTiff")
```

And the notebook can do this:

```
!SAR2D2_ENV={env} ./run.sh sardem --bbox {bbox} --data-source COP -o {dem_file} --output-format GTiff
```